Tests/Marketplace/marketplace_services.py

```diff
@@ -1628,60 +1642,61 @@ def _enhance_pack_attributes(self, user_metadata, index_folder_path, pack_was_mo
-        self._is_premium = self._is_private_pack
-        self._preview_only = get_valid_bool(user_metadata.get('previewOnly', False))
-        self._price = convert_price(pack_id=self._pack_name, price_value_input=user_metadata.get('price'))
-        if self._is_private_pack:
-            self._vendor_id = user_metadata.get('vendorId', "")
-            self._partner_id = user_metadata.get('partnerId', "")
-            self._partner_name = user_metadata.get('partnerName', "")
-            self._content_commit_hash = user_metadata.get('contentCommitHash', "")
-            # Currently all content packs are legacy.
-            # Since premium packs cannot be legacy, we directly set this attribute to false.
-            self._legacy = False
-
-        # ===== Pack Statistics Attributes =====
-        if not self._is_private_pack and statistics_handler:  # Public Content case
-            self._pack_statistics_handler = mp_statistics.PackStatisticsHandler(
-                self._pack_name, statistics_handler.packs_statistics_df, statistics_handler.packs_download_count_desc,
-                displayed_dependencies
-            )
-            self._downloads_count = self._pack_statistics_handler.download_count
-            trending_packs = statistics_handler.trending_packs
-            pack_dependencies_by_download_count = self._pack_statistics_handler.displayed_dependencies_sorted
+        if not only_dependencies_reformat:
+            self._is_private_pack = 'partnerId' in user_metadata
+            self._is_premium = self._is_private_pack
+            self._preview_only = get_valid_bool(user_metadata.get('previewOnly', False))
+            self._price = convert_price(pack_id=self._pack_name, price_value_input=user_metadata.get('price'))
+            if self._is_private_pack:
+                self._vendor_id = user_metadata.get('vendorId', "")
+                self._partner_id = user_metadata.get('partnerId', "")
+                self._partner_name = user_metadata.get('partnerName', "")
+                self._content_commit_hash = user_metadata.get('contentCommitHash', "")
+                # Currently all content packs are legacy.
+                # Since premium packs cannot be legacy, we directly set this attribute to false.
+                self._legacy = False
+
+            # ===== Pack Statistics Attributes =====
+            if not self._is_private_pack and statistics_handler:  # Public Content case
+                self._pack_statistics_handler = mp_statistics.PackStatisticsHandler(
+                    self._pack_name, statistics_handler.packs_statistics_df, statistics_handler.packs_download_count_desc,
+                    displayed_dependencies
+                )
+                self._downloads_count = self._pack_statistics_handler.download_count
+                trending_packs = statistics_handler.trending_packs
+                pack_dependencies_by_download_count = self._pack_statistics_handler.displayed_dependencies_sorted
 
```

Comment:
You need this also for `get_all_pack_images` and for that, you will need the PackStatisticsHandler, make sure you are not initializing it again